clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -306,6 +306,34 @@ static int64_t getIntExprValue(const Expr *E, ASTContext &Ctx) {
   return E->getIntegerConstantExpr(Ctx)->getSExtValue();
 }
 
+// Collect function attributes related to SYCL
+static void collectSYCLAttributes(Sema &S, FunctionDecl *FD,
+                                  llvm::SmallVector<Attr *, 4> &Attrs,
+                                  bool DirectlyCalled = true) {
+  if (!FD->hasAttrs())
+    return;
+  for (Attr *A : FD->getAttrs()) {
```

Comment:
Given the switch from `SmallPtrSet` to `SmallVector`, this can be switched to an `llvm::copy_if()` (using a `std::back_inserter()`).

Reply:
OK, done.